core/src/test/java/io/el/concurrent/DefaultEventLoopGroupTest.java

```diff
@@ -0,0 +1,279 @@
+
+    private final CountDownLatch threadLock = new CountDownLatch(1);
+
+    private volatile State state = State.NOT_STARTED;
+
+    public MockEventLoop(Executor executor) {
+      super(executor);
+    }
+
+    @Override
+    protected void run() {
+      while (!confirmShutdown()) {
+        Runnable task = takeTask();
+        if (task != null) {
+          task.run();
+        }
+      }
+    }
+
+    @Override
+    public void execute(Runnable task) {
+      checkNotNull(task, "task");
+      if (inEventLoop()) {
+        return;
+      }
+      if (!state.equals(State.NOT_STARTED)) {
+        return;
+      }
+      if (!stateUpdater.compareAndSet(this, State.NOT_STARTED, State.STARTED)) {
+        return;
+      }
+      boolean success = false;
+      try {
+        doStart();
+        success = true;
+      } finally {
+        if (!success) {
+          stateUpdater.compareAndSet(this, State.STARTED, State.TERMINATED);
+        }
+      }
+    }
+
+    private void doStart() {
+      try {
+
+      } finally {
+        stateUpdater.compareAndSet(this, State.STARTED, State.TERMINATED);
+      }
+    }
+
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) {
+      if (inEventLoop()) {
+        throw new IllegalStateException("cannot await termination of the current thread");
+      }
+
+      try {
+        threadLock.await(timeout, unit);
+      } catch (InterruptedException e) {
+      }
+      return isTerminated();
+    }
+
+    @Override
+    public boolean isTerminated() {
+      return state.equals(State.TERMINATED);
+    }
+  }
+
+  private static class MockDefaultEventLoopGroup extends DefaultEventLoopGroup {
+
+    protected MockDefaultEventLoopGroup(int nThreads, Executor executor,
+        EventLoopChooserFactory chooserFactory) {
+      super(nThreads, executor, chooserFactory);
+    }
+
+    @Override
+    public void shutdown() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks, long timeout,
+        TimeUnit unit) throws InterruptedException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks)
+        throws InterruptedException, ExecutionException {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+        throws InterruptedException, ExecutionException, TimeoutException {
+      throw new UnsupportedOperationException();
+    }
```

Comment:
It would be great if all of these unsupported methods are implemented in `DefaultEventLoopGroup` not in mock

Reply:
done in oral discussion.